Source/GameView/Repositories/GameAPIRepo.swift

```diff
@@ -78,58 +78,81 @@ public class GameAPIRepo: GameRepository {
     
   }
   
-  //TODO: Replace gameList with individual game and make the model call the function many times.
   public func getGameDetails(of gameList: [Game], with serviceCaller: ServiceCaller) {
     
-    let dispatchGroup = DispatchGroup()
+    let operationQueue = OperationQueue()
     
-    gameList.forEach({ (game) in
-      
-      dispatchGroup.enter()
+    let serviceCallOperation = createServiceCallBlockOperation(using: serviceCaller,
+                                                               forGamesList: gameList)
+    
+    let notifyOperation = BlockOperation {
+      self.notifyGameDetailsLoaded(withData: gameList)
+    }
+    
+    notifyOperation.addDependency(serviceCallOperation)
+    
+    operationQueue.addOperation(serviceCallOperation)
+    operationQueue.addOperation(notifyOperation)
+    
+  }
+  
+  private func createServiceCallBlockOperation(using serviceCaller: ServiceCaller,
+                                               forGamesList gameList: [Game]) -> BlockOperation {
+    
+    let serviceCallOperation = BlockOperation {
       
-      let dataBundle = DataBundle()
-      dataBundle.extraData["game"] = game
+      let dispatchGroup = DispatchGroup()
       
-      // MARK: - Call Succeeded Closure
-      serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+      gameList.forEach({ (game) in
+        
+        dispatchGroup.enter()
         
-        guard let game = dataBundle.extraData["game"] as? Game else {
-          return
+        let dataBundle = DataBundle()
+        dataBundle.extraData["game"] = game
+        
+        // MARK: - Call Succeeded Closure
+        serviceCaller.callSucceeded = { [weak self] data, dataBundle in
+          
+          guard let game = dataBundle.extraData["game"] as? Game else {
+            return
+          }
+          
+          guard let self = self else {
```

Comment:
Two things here, rather have your self unwrapped at the top so that if self is nil we exit immediately
Secondly, I have not used dispathGroup that often but shouldn't you want to leave the group even if we do an early return ?

Reply:
Yeah, the early return should include the leave. I'll have to add that. Moving the unwrapping of self to the top also makes sense, otherwise there could be a lot of processing happening only for it to be chucked out the window